### PR TITLE
lsp-xml: Allow automatic download of Lemminx (LSP XML) server

### PR DESCRIPTION
**Pull request by egh**
- Add helper constants to reduce repetition of current version.
- Store jar in `lsp-server-install-dir`

### CODE REVIEW

File: clients/lsp-xml.el [hunk 1]
```diff
@@ -182,20 +182,33 @@ Newlines and excess whitespace are removed."
   ("xml.catalogs" lsp-xml-catalogs)
   ("xml.trace.server" lsp-xml-trace-server)))
 
-(defcustom lsp-xml-jar-file (expand-file-name
-                             (locate-user-emacs-file
-                              "org.eclipse.lsp4xml-0.3.0-uber.jar"))
+(defconst lsp-xml-jar-version "0.13.1")
+
+(defconst lsp-xml-jar-name (s-concat "org.eclipse.lemminx-" lsp-xml-jar-version "-uber.jar"))
```
Comment (kiennq) [inline clients/lsp-xml.el:187]: nit: a format would look easier
```suggestion
(defconst lsp-xml-jar-name (format "org.eclipse.lemminx-%s-uber.jar" lsp-xml-jar-version))
```

File: clients/lsp-xml.el [hunk 2]
```diff
@@ -182,20 +182,33 @@ Newlines and excess whitespace are removed."
...
+
+(defcustom lsp-xml-jar-file (f-join lsp-server-install-dir "xmlls" lsp-xml-jar-name)
   "Xml server jar command."
   :type 'string
   :group 'lsp-xml
   :type 'file
   :package-version '(lsp-mode . "6.1"))
 
+(defcustom lsp-xml-jar-download-url
+  (s-concat
```
Comment (kiennq) [inline clients/lsp-xml.el:197]: Same here, `format` is easier to see how the final url looks like

File: clients/lsp-xml.el (hunk 1 above)
Comment (egh) [reply clients/lsp-xml.el]: Thank you! Fixed in 73442c84

File: clients/lsp-xml.el (hunk 2 above)
Comment (egh) [reply clients/lsp-xml.el]: 73442c84

**Comment by kiennq** [comment]
Just asking but is `Lemminx` server is newer (maintained) version of `lsp4xml`?

**Comment by egh** [comment]
> Just asking but is Lemminx server is newer (maintained) version of lsp4xml?

It seems the project was renamed and transferred to the Eclipse project.
See https://github.com/eclipse/lemminx/issues/283 and https://projects.eclipse.org/projects/technology.lemminx
https://github.com/angelozerr/lsp4xml now redirects to https://github.com/eclipse/lemminx

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Comment by egh** [comment]
Thank you!

**Pull request merged**